infra/ansible/config/cuda_deps.yaml

```diff
@@ -3,12 +3,14 @@
 cuda_deps:
   # List all libcudnn8 versions with `apt list -a libcudnn8`
   libcudnn:
+    "12.3": libcudnn9-cuda-12=9.0.0.312-1
```

Comment:
I wonder where you got the `libcudnn9-cuda-12=9.0.0.312-1`

Reply:
Just searching for libcudnn9 versions doesn't give any results using apt list -a libcudnn9. To install cudnn9 nvidia recommends `sudo apt-get -y install cudnn9-cuda-12` 
![image](https://github.com/pytorch/xla/assets/15787503/c2fcaf36-2414-4aa7-9ab5-466d3fd45876)

So searching for apt list -a libcudnn9-cuda-12 gave the above version.